blogs/posts/2023-04-26-reinstalling-r-packages.qmd

```diff
@@ -0,0 +1,31 @@
+---
+title: "Reinstalling R Packages"
+author: "Tom Jemmett"
+date: "2023-04-26"
+categories: [git, tutorial]
+knitr:
+  opts_chunk:
+    fig.height: 2.5
+    echo: false
+    dev: png
+    dev.args:
+      bg: transparent
+---
+
+[R 4.3.0 was released](https://stat.ethz.ch/pipermail/r-announce/2023/000691.html) last week. Anytime you update R you will probably find yourself in the position where no packages are installed. This is by design - the packages that you have installed may need to be updated and recompiled to work under new versions of R.
+
+You may find yourself wanting to have all of the packages that you previously used, so one approach that some people take is to copy the previous library folder to the new versions folder. This isn't a good idea and could potentially break your R install.
+
+Another approach would be to export the list of packages in R before updating and then using that list after you have updated R. This can cause issues though if you install from places other than CRAN, e.g. bioconductor, or from GitHub.
+
+Some of these approaches are discussed on the [RStudio Community Forum](https://community.rstudio.com/t/reinstalling-packages-on-new-version-of-r/7670/4). But I prefer an approach of having a "spring clean", instead only installing the packages that I know that I need.
+
+I maintain a [list of the packages that I used](https://gist.github.com/tomjemmett/c105d3e0fbea7558088f68c65e68e1ed/) as a gist. Using this, I can then simply run this script on any new R install. In fact, if you click the "raw" button on the gist, and copy that url, you can simply run
+
+``` r
+source("https://gist.githubusercontent.com/tomjemmett/c105d3e0fbea7558088f68c65e68e1ed/raw/a1db4b5fa0d24562d16d3f57fe8c25fb0d8aa53e/setup.R")
+```
+
+to run the script. Generally, sourcing a url is a bad idea - the reason for this is if it's not a link that you control, then someone could update the contents and run arbritary code on your machine. In this case, I'm happy to run this as it's my own gist, but you should be mindful if running it yourself!
+
+If you look at the script I first install a number of packages from CRAN, then I install packages that only exist on GitHub.
```

Comment:
I've updated today and realised that it goes to another folder so I still have access to 4.2.2 through Tools/Global Options and change the R version. It means I still have access to all or my old installed packages even though I forgot to look what I should note down to follow this!